commit aad6e5723acbeaa3281c26ba07d389b3f73fb5b7 (refs/changes/34/2134/4)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2017-07-13T11:29:50+00:00 (2 years, 3 months ago)
    
    SAPI-290 getApplicationObjects returns undefined instance list for known services
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Trent Mick <trent.mick@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 4423017..6d3b35c 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,16 +4,16 @@
     file, You can obtain one at http://mozilla.org/MPL/2.0/.
 -->
 
-<!--
-    Copyright 2016 Joyent, Inc.
--->
-
 # sdc-clients Changelog
 
 ## not yet released
 
 (nothing yet)
 
+## 10.2.1
+
+- SAPI-290 getApplicationObjects returns undefined instance list for known services
+
 ## 10.2.0
 
 - PUBAPI-1380 Cloudapi should support wildcards in ListPackages
diff --git a/lib/sapi.js b/lib/sapi.js
index 0cdaa7b..c82ff39 100644
--- a/lib/sapi.js
+++ b/lib/sapi.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -752,16 +752,16 @@ function getApplicationObjects(app_uuid, opts, cb) {
                     return (subcb(err));
 
                 ret.services = {};
+                ret.instances = {};
                 svcs.forEach(function (svc) {
                     ret.services[svc.uuid] = svc;
+                    ret.instances[svc.uuid] = [];
                 });
 
                 return (subcb());
             });
         },
         function (subcb) {
-            ret.instances = {};
-
             var search_opts = {};
             if (opts.include_master)
                 search_opts.include_master = true;
@@ -780,8 +780,6 @@ function getApplicationObjects(app_uuid, opts, cb) {
                     var svc_uuid = inst.service_uuid;
                     if (!ret.services[svc_uuid])
                         return;
-                    if (!ret.instances[svc_uuid])
-                        ret.instances[svc_uuid] = [];
                     ret.instances[svc_uuid].push(inst);
                 });
 
diff --git a/package.json b/package.json
index 10d796e..074c6b7 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdc-clients",
   "description": "node.js client libraries for Triton core REST APIs.",
-  "version": "10.2.0",
+  "version": "10.2.1",
   "homepage": "http://www.joyent.com",
   "repository": {
     "type": "git",
